src/server/pfs/server/transaction_defer.go

```diff
@@ -56,3 +56,50 @@ func (t *Propagater) Run(ctx context.Context) error {
 	}
 	return t.d.propagateBranches(ctx, t.txnCtx, branches)
 }
+
+type RepoChecker struct {
+	d      *driver
+	txnCtx *txncontext.TransactionContext
+	repos  []*pfs.Repo
```

Comment:
Lets make this a map from the repo key to the repo proto. The same repo can be added multiple times.